app/controllers/criteria_controller.rb

```diff
@@ -0,0 +1,32 @@
+class CriteriaController < ApplicationController
+
+  def new
+    @assignment = Assignment.find(params[:assignment_id])
+    @criterion_type = params[:criterion_type]
+    @criterion = get_class(@criterion_type).new
```

Comment:
Yes, I was thinking that would interfere with the string polarization we use for controllers, but I just saw the tableize method. Thank you for your advice.